pandas/tests/test_algos.py

```diff
@@ -841,6 +840,13 @@ def test_same_nan_is_in(self):
         result = algos.isin(comps, values)
         tm.assert_numpy_array_equal(expected, result)
 
+    def test_same_nan_is_in_large(self):
```

Comment:
It's somewhat duplicative, but could you also add a test for Series to show that the specific behavior from the bug report is fixed?